build_tools/math/generate_tests.py

```diff
@@ -62,11 +70,9 @@ def main():
     return
 
   fa_version = tuple(map(int, fa.__version__.split(".", 4)[:3]))
-  if fa_version < (0, 4, 0):
-    warnings.warn(
-        "functional_algorithm version 0.4.0 or newer is required,"
-        f" got {fa.__version__}"
-    )
+  if fa_version < (0, 5, 0):
```

Comment:
No, it would not be wrong per se. In this particular case, notice that the scripts generate_ChloDecompositionPatternsMath.py and generate_tests.py are independent and these use independent parts of FA (for instance, the first script relies on FA code generation while the second uses FA interface to mpmath). The whole point of having these versions checks is to ensure that whoever runs the scripts, must have the latest version of FA installed in the build environment (FA version changes rather quickly because when a new algorithm is added to FA, a new release of FA is created).

Another way to enforce FA version requirements would be to have a helper Python module that implements the version checks and both `generate_*.py` would import it.

Reply:
The required version of functional_algorithms package is now defined in a single location `README.md` which both generate scripts scan.